homeassistant/components/sonarr/__init__.py

```diff
@@ -106,6 +112,17 @@ async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> boo
     return unload_ok
 
 
+def _async_start_reauth(hass: HomeAssistantType, entry: ConfigEntry):
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_REAUTH},
+            data={"config_entry_id": entry.entry_id, **entry.data},
+        )
+    )
+    _LOGGER.error("API Key is no longer valid. Please reauthenticate")
```

Comment:
I wonder if we should log it like this. Basically, we expect a user to take action, just raising an error in the log doesn't feel user-friendly to me.

Instead, should be raise a persistent notification in these cases and dismiss those when the re-auth it done? 

I did a similar approach in the Spotify integration for this. Not sure though, as we do not have a policy on this.

Reply:
right ive seen it both ways. i guess a notice wouldnt hurt